### PR TITLE
Enable pre- and post-scp commands on remote host

### PR DESCRIPTION
**Pull request by jan-brinkmann**
This enables us to execute commands on the remote host where we transfer a backup to.

When we transfer a backup by means of scp, we probably have access to execute commands on that host. Now, we can overhand arbitrary commands in the environmental variables `PRE_SCP_COMMAND` and `POST_SCP_COMMAND` that are executet before and after the backup is transferred.

There is an example `docker-compose.yml` file in `tests/pre-post-scp-command` showing how `rotate-backups` can be executed after the backup has been transferred. Additionally, there is a reference in the `README.md`.

### CODE REVIEW

File: src/backup.sh [hunk 1]
```diff
@@ -121,11 +121,20 @@ fi
 
 if [ ! -z "$SCP_HOST" ]; then
   info "Uploading backup by means of SCP"
+  SSH_CONFIG="-o StrictHostKeyChecking=no -i /ssh/id_rsa"
+  if [ ! -z "$PRE_SCP_COMMAND" ]; then
+      echo "Pre-scp command: $PRE_SCP_COMMAND"
+      ssh $SSH_CONFIG $SCP_USER@$SCP_HOST $PRE_SCP_COMMAND
```
Comment (jareware) [inline src/backup.sh:127]:
```suggestion
    echo "Pre-scp command: $PRE_SCP_COMMAND"
    ssh $SSH_CONFIG $SCP_USER@$SCP_HOST $PRE_SCP_COMMAND
```

File: src/backup.sh [hunk 2]
```diff
@@ -121,11 +121,20 @@ fi
...
+  fi
   echo "Will upload to $SCP_HOST:$SCP_DIRECTORY"
   TIME_UPLOAD="$(date +%s.%N)"
-  scp -ro StrictHostKeyChecking=no -i /ssh/id_rsa $BACKUP_FILENAME $SCP_USER@$SCP_HOST:$SCP_DIRECTORY
+  scp $SSH_CONFIG $BACKUP_FILENAME $SCP_USER@$SCP_HOST:$SCP_DIRECTORY
   echo "Upload finished"
   TIME_UPLOADED="$(date +%s.%N)"
+  if [ ! -z "$POST_SCP_COMMAND" ]; then
+      echo "Post-scp command: $POST_SCP_COMMAND"
+      ssh $SSH_CONFIG $SCP_USER@$SCP_HOST $POST_SCP_COMMAND
```
Comment (jareware) [inline src/backup.sh:136]:
```suggestion
    echo "Post-scp command: $POST_SCP_COMMAND"
    ssh $SSH_CONFIG $SCP_USER@$SCP_HOST $POST_SCP_COMMAND
```

**Review by jareware — APPROVED** [review]
Beautiful, thank you! 👏👏

**Comment by jareware** [comment]
@jan-brinkmann this is now in `2.5.0` which was just pushed, want to pull it and check that it works for your use case?

**Comment by jan-brinkmann** [comment]
@jareware, yes, it does!
Thank you!

**Comment by jareware** [comment]
Thanks again for a great contribution!

**Pull request merged**